### PR TITLE
Update to always return AMP_MODE_SECONDARY.

### PR DESCRIPTION
**Pull request by Hazlitte**
## Summary

Update to always return AMP_MODE_SECONDARY.

Addresses issue #2998

## Checklist

- [X] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [X] I have added a QA Brief on the issue linked above.
- [X] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

**Comment by tofumatt** [comment]
@Hazlitte Just a reminder to flag the PRs as "ready for review" once they're moved into the Code Review column, or the E2E tests and many other checks won't run. Thanks!

**Review by tofumatt — CHANGES_REQUESTED** [review]
Looks like the E2E AMP tests are failing as we now should never return `AMP_MODE_PRIMARY` as a mode.

I'm a bit surprised that this test isn't failing: https://github.com/google/site-kit-wp/blob/80868c0780a1d9febc153adadea50617624a22db/tests/phpunit/includes/TestCase.php#L166-L179

Given, from what I can tell, we should never return this mode anymore: https://github.com/google/site-kit-wp/blob/80868c0780a1d9febc153adadea50617624a22db/includes/Context.php#L28-L34

In fact, it might be worth adding a comment to the `const AMP_MODE_PRIMARY = 'primary';` declaration that this mode is currently unused because of this issue.

**Review by tofumatt — CHANGES_REQUESTED** [review]
There are a few tests that reference `get_amp_primary_context`, which we no longer use. I suppose they're fine to leave in, but just noting it.

Please test these changes with and without the AMP plugin installed, as when it's installed and enabled an error is thrown.

It would be good to test with a pre-2.0 version of the AMP plugin as well, obtained from https://wordpress.org/plugins/amp/advanced/

File: includes/Context.php [hunk 1]
```diff
@@ -323,10 +323,6 @@ public function get_amp_mode() {
 			return false;
 		}
 
-		$exposes_support_mode = defined( 'AMP_Theme_Support::STANDARD_MODE_SLUG' )
-			&& defined( 'AMP_Theme_Support::TRANSITIONAL_MODE_SLUG' )
-			&& defined( 'AMP_Theme_Support::READER_MODE_SLUG' );
-
```
[the remark this answers is not in the document]
Comment (Hazlitte) [reply includes/Context.php:329]: @tofumatt Thank you, I didn't spot that.

**Pull request closed without merging**